guides/v2.2/frontend-dev-guide/layouts/xml-manage.md

```diff
@@ -345,56 +344,143 @@ This would make the product page look like following:
 To learn how to locate the layout file you need to customize, see [Locate templates, layouts, and styles].
 
 
-## Remove elements {#layout_markup_remove_elements}
+## Add functionality to existing elements {#layout_markup_add_to_elements}
 
-Elements are removed using the `remove` attribute for the `<referenceBlock>` and `<referenceContainer>`. 
+Let us say that we want to add functionality to a core template with custom logic using a ViewModel in the `cart/item/default.phtml` template found in `Magento/Checkout/view/frontend/layout/checkout_cart_item_renderers.xml`:
 
-**Example**: remove the Compare Products {% glossarytooltip 31751771-8163-434b-88bc-c5f94d859fc3 %}sidebar{% endglossarytooltip %} block from all store pages. 
+```xml
+<?xml version="1.0"?>
+<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
+<body>
+    <referenceBlock cart.renderer="Magento/Checkout/view/frontend/layout/checkout_cart_item_renderers.xml">
+        <arguments>
+           <argument name="viewModel" xsi:type="object">Vendor\CustomModule\ViewModel\Class</argument>
+        </arguments>
+    </referenceBlock>
+</body>
+```
+
+You would also have to implement the right interface in your viewModel class (i.e. `ArgumentInterface`):
+
+```php
+namespace Vendor\CustomModule\ViewModel;
+
+class Class implements \Magento\Framework\View\Element\Block\ArgumentInterface
+{
+    public function __construct()
+    {
 
-This block is declared in `app/code/Magento/Catalog/view/frontend/layout/default.xml`:
+    }
+}
+```
+## Modify functionality with plugins (interceptors) {#layout_markup_modify_with_plugins}
+
+To substitute or extend the behavior of original, public methods for any class or interface, we can make use of plugins, or interceptors, which are classes that modify the behavior of public class functions by intercepting a function call and running code before, after, or around that function call in the form of listeners. 
+
+This interception approach reduces conflicts among extensions that change the behavior of the same class or method, with a Plugin class implementation changing only the behavior of a class function, rather than overriding the entire class.
+
+In order to use plugins (interceptors), we must first define them in the di.xml of the module:
 
 ```xml
-<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
-    <body>
-...
-        <referenceContainer name="sidebar.additional">
-            <block class="Magento\Catalog\Block\Product\Compare\Sidebar" name="catalog.compare.sidebar" template="product/compare/sidebar.phtml"/>
-        </referenceContainer>
-...
-    </body>
-</page>
+<config>
+    <type name="{ObservedType}">
+      <plugin name="{pluginName}" type="{PluginClassName}" sortOrder="1" disabled="false" />
```

Comment:
I suggest removing `sortOrder` and `disabled` attributes. They should not be encouraged or implied as necessary.